compiler-cli/src/publish.rs

```diff
@@ -254,6 +256,26 @@ core team.\n",
     Ok(should_publish)
 }
 
+fn check_for_git_dependencies(config: &PackageConfig) -> Result<bool, Error> {
+    let contains_git_dependencies = |dependencies: &HashMap<EcoString, Requirement>| {
+        dependencies
+            .iter()
+            .any(|(_, requirement)| matches!(requirement, Requirement::Git { .. }))
+    };
+
+    if contains_git_dependencies(&config.dependencies)
+        || contains_git_dependencies(&config.dev_dependencies)
+    {
+        println!(
+            "You are trying to publish a package which uses git dependencies.
+Git dependencies cannot be used in published packages."
+        );
+        Ok(false)
+    } else {
+        Ok(true)
+    }
+}
+
```

Comment:
I'd change this so that it produces the same nice error used by the path dependencies check:
![Screenshot 2025-02-16 alle 13 31 56](https://github.com/user-attachments/assets/e1dd17b3-1cae-4bb4-ad52-487b7610cd49)